controllers/common/supportingservice_controller.go

```diff
@@ -0,0 +1,131 @@
+//+kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;create;list;watch;delete;update
+//+kubebuilder:rbac:groups=core,resources=configmaps;events;pods;secrets;services,verbs=create;delete;get;list;patch;update;watch
+
+// Reconcile reads that state of the cluster for a KogitoSupportingService object and makes changes based on the state read
+// and what is in the KogitoSupportingService.Spec
+func (r *KogitoSupportingServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, resultErr error) {
+	log := logger.FromContext(ctx)
+	log.Info("Reconciling for KogitoSupportingService")
+
+	// create kogitoContext
+	kogitoContext := operator.Context{
+		Client:  r.Client,
+		Log:     log,
+		Scheme:  r.Scheme,
+		Version: app2.Version,
+	}
+
+	// Fetch the KogitoSupportingService instance
+	supportingServiceHandler := r.SupportingServiceHandler(kogitoContext)
+	instance, resultErr := supportingServiceHandler.FetchKogitoSupportingService(req.NamespacedName)
+	if resultErr != nil {
+		return
+	}
+	if instance == nil {
+		log.Debug("kogitoSupportingService Instance not found")
+		return
+	}
+
+	supportingServiceManager := manager.NewKogitoSupportingServiceManager(kogitoContext, supportingServiceHandler)
+	if resultErr = supportingServiceManager.EnsureSingletonService(req.Namespace, instance.GetSupportingServiceSpec().GetServiceType()); resultErr != nil {
+		return
+	}
+
+	runtimeHandler := r.RuntimeHandler(kogitoContext)
+	infraHandler := r.InfraHandler(kogitoContext)
+	reconcileHandler := kogitosupportingservice.NewReconcilerHandler(kogitoContext, infraHandler, supportingServiceHandler, runtimeHandler)
+	reconciler := reconcileHandler.GetSupportingServiceReconciler(instance)
+	resultErr = reconciler.Reconcile()
+	if resultErr != nil {
+		return infrastructure.NewReconciliationErrorHandler(kogitoContext).GetReconcileResultFor(resultErr)
+	}
+	return
+}
+
+// SetupWithManager registers the controller with manager
+func (r *KogitoSupportingServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	pred := predicate.Funcs{
+		// Don't watch delete events as the resource removals will be handled by its finalizer
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetDeletionTimestamp().IsZero()
+		},
+	}
+
+	b := ctrl.NewControllerManagedBy(mgr).
+		For(r.ReconcilingObject, builder.WithPredicates(pred)).
+		Owns(&corev1.Service{}).Owns(&appsv1.Deployment{}).Owns(&corev1.ConfigMap{})
+
+	//infraHandler := &handler.EnqueueRequestForOwner{IsController: false, OwnerType: &appv1beta1.KogitoSupportingService{}}
+	//infraPred := predicate.Funcs{
+	//	UpdateFunc: func(e event.UpdateEvent) bool {
+	//		return reflect.DeepEqual(e.ObjectNew.GetOwnerReferences(), e.ObjectOld.GetOwnerReferences())
+	//	},
+	//}
+	//b.Watches(&source.Kind{Type: &appv1beta1.KogitoInfra{}}, infraHandler, builder.WithPredicates(infraPred))
```

Comment:
same here

Reply:
Its not required anymore because I had removed KogitoInfra ownership from KogitoRuntime and KogitoSupportingServices.